drivers/sensor/adxl372/adxl372.c

```diff
@@ -777,18 +670,74 @@ static int adxl372_probe(const struct device *dev)
 		return -ENODEV;
 	}
 
-#ifdef CONFIG_ADXL372_I2C
-	/*
-	 * When sharing an SDA bus, the ADXL372 Silcon REV < 3  may prevent
-	 * communication with other devices on that bus.
-	 */
-	adxl372_reg_read(dev, ADXL372_REVID, &dev_id);
-	if (dev_id < 3) {
-		LOG_WRN("The ADXL372 Rev %u only supports point to point I2C communication!",
-			    dev_id);
-	}
+	data->max_peak_detect_mode = IS_ENABLED(CONFIG_ADXL372_PEAK_DETECT_MODE),
+
+#ifdef CONFIG_ADXL372_ODR_400HZ
+	data->odr = ADXL372_ODR_400HZ,
+#elif CONFIG_ADXL372_ODR_800HZ
+	data->odr = ADXL372_ODR_800HZ,
+#elif CONFIG_ADXL372_ODR_1600HZ
+	data->odr = ADXL372_ODR_1600HZ,
+#elif CONFIG_ADXL372_ODR_3200HZ
+	data->odr = ADXL372_ODR_3200HZ,
+#elif CONFIG_ADXL372_ODR_6400HZ
+	data->odr = ADXL372_ODR_6400HZ,
 #endif
 
+#ifdef CONFIG_ADXL372_BW_200HZ
+	data->bw = ADXL372_BW_200HZ,
+#elif CONFIG_ADXL372_BW_400HZ
+	data->bw = ADXL372_BW_400HZ,
+#elif CONFIG_ADXL372_BW_800HZ
+	data->bw = ADXL372_BW_800HZ,
+#elif CONFIG_ADXL372_BW_1600HZ
+	data->bw = ADXL372_BW_1600HZ,
+#elif CONFIG_ADXL372_BW_3200HZ
+	data->bw = ADXL372_BW_3200HZ,
+#elif CONFIG_ADXL372_LPF_DISABLE
+	data->bw = ADXL372_BW_LPF_DISABLED,
+#endif
+
+#ifdef CONFIG_ADXL372_HPF_CORNER0
+	data->hpf = ADXL372_HPF_CORNER_0,
+#elif CONFIG_ADXL372_HPF_CORNER1
+	data->hpf = ADXL372_HPF_CORNER_1,
+#elif CONFIG_ADXL372_HPF_CORNER2
+	data->hpf = ADXL372_HPF_CORNER_2,
+#elif CONFIG_ADXL372_HPF_CORNER3
+	data->hpf = ADXL372_HPF_CORNER_3,
+#elif CONFIG_ADXL372_HPF_DISABLE
+	data->hpf = ADXL372_HPF_DISABLED,
+#endif
```

Comment:
These options should be converted to devicetree properties to allow configuring them on a per-instance basis.